pkg/registries/adapters/openshift_adapter.go

```diff
@@ -83,17 +82,40 @@ func (r OpenShiftAdapter) getOpenShiftAuthToken() (string, error) {
 	}
 	username := r.Config.User
 	password := r.Config.Pass
-	authString := fmt.Sprintf("%v:%v", username, password)
 
-	authString = b64.StdEncoding.EncodeToString([]byte(authString))
+	req, err := http.NewRequest("GET", fmt.Sprintf("%v/v2/", r.Config.URL), nil)
+	if err != nil {
+		return "", err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// Ensure that response holds data we expect
+	if resp.Header["Www-Authenticate"] == nil {
+		return "", errors.New("failed to find www-authenticate header from response")
+	}
 
-	req, err := http.NewRequest("GET", openShiftAuthURL, nil)
+	authChallenge := resp.Header["Www-Authenticate"][0]
+	if strings.Index(authChallenge, "realm=\"") == -1 {
+		return "", errors.New("failed to find realm in www-authenticate header")
+	}
+	authRealm := strings.Split(strings.Split(authChallenge, "realm=\"")[1], "\"")[0]
+	authOptions := strings.Split(authChallenge, ",")[1]
```

Comment:
I don't think there is a check for `,` in the auth challenge. this might panic if we got bad data for the header